w3af/plugins/crawl/find_backdoors.py

```diff
@@ -118,27 +118,30 @@ def _check_if_exists(self, web_shell_url):
         except BaseFrameworkException:
             om.out.debug('Failed to GET webshell:' + web_shell_url)
         else:
-            signature = self._match_signature(response)
-            if signature is None:
-                return
+            if response.get_code() == 200:
+                signature = self._match_signature(response)
+                if signature is None:
+                    return
```

Comment:
Yes, this is the only functional change. The motivation was 100% false positives on any 404 on a clean install of tomcat as in the message (included in screenshot) it will contain the words you are looking for. It has a nice side benefit of performance but that was not the motivation.

Hmm thats an interesting thought about a backdoor returning a 404 but I dont see any way to get around that with the current implementation and would a massive change in order to discover and would probably need some kind of server access to really truly vet its present and they are clever like that.